web/static/js/thesis-editor.js

```diff
@@ -56,25 +74,61 @@ class ThesisEditor extends React.Component {
   }
 
   pageDescription () {
-    const desc = this.descriptionMetaTag()
+    const desc = this.pageDescriptionMetaTag()
     return desc ? desc.content : null
   }
 
-  descriptionMetaTag () {
+  pageRedirectURL () {
+    return thesisContainer.getAttribute('data-redirect-url')
+  }
+
+  pageTemplate () {
+    return thesisContainer.getAttribute('data-template')
+  }
+
+  pageTemplates () {
+    return thesisContainer.getAttribute('data-templates').split(",")
+  }
+
+  pageDescriptionMetaTag () {
     return document.querySelectorAll('meta[name=description]')[0]
   }
 
   trayCanceled () {
     this.setState({trayOpen: false, trayData: null})
   }
 
-  settingsTraySubmitted (page) {
-    document.title = page.title
+  settingsTraySubmitted (data) {
```

Comment:
If we're creating a new page, submit the info to the server and redirect to it. If we're just editing the current page, wait for the user to actually save before doing that.